umbraco/Umbraco.CMS.Backoffice
# Feature: Entity Is trashed condition + filter document/media entity actions

## Description

Filter entity actions based on the entity is trashed or not. The feature is implemented for both document and media. The PR adds conditions to all entity and workspace actions.


**Document**
![Screenshot 2024-05-02 at 10 27 35](https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/6078361/1c3f578d-4625-4015-91f6-1f947af144c7)

**Document in Recycle Bin**
![Screenshot 2024-05-02 at 10 26 24](https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/6078361/a01f215b-58a2-4cb1-9c45-c889979526d3)


## How to test:
* Make sure only the allowed entity actions are available on documents
* Make sure only the allowed entity actions are available on documents in the recycle bin
* Same tests for media.

## Types of changes
